src/swaps/hop/HopSwapProvider.ts

```diff
@@ -0,0 +1,579 @@
+      [Hop.Chain.Ethereum.slug]: Hop.Chain.Ethereum,
+      [Hop.Chain.Arbitrum.slug]: Hop.Chain.Arbitrum,
+      [Hop.Chain.Gnosis.slug]: Hop.Chain.Gnosis,
+      [Hop.Chain.Optimism.slug]: Hop.Chain.Optimism,
+      [Hop.Chain.Polygon.slug]: Hop.Chain.Polygon
+    }
+    return slugToChain[chainName]
+  }
+  // L2->L1 or L2->L2
+  GQL_getDestinationTxHashFromL2Source(transferId: string) {
+    return `query {
+        withdrawalBondeds(
+          where: {
+            transferId: "${transferId}"
+          }
+        ) {
+          timestamp
+          amount
+          transactionHash
+          token
+          timestamp
+        }
+      }
+    `
+  }
+  // L1->L2
+  GQL_getDestinationTxHashFromL1Source(recipient : string) {
+    return `query {
+        transferFromL1Completeds(
+          where: {
+            recipient: "${recipient}"
+          },
+          orderBy: timestamp,
+          orderDirection: desc
+        ) {
+          timestamp
+          amount
+          transactionHash
+          token
+          timestamp
+        }
+      }
+    `
+  }
+
+  GQL_getTransferIdByTxHash(txHash: string) {
+    return `query {
+        transferSents(
+          where: {
+            transactionHash: "${txHash}"
+          }
+        ) {
+          timestamp
+          transferId
+          amount
+          bonderFee
+          transactionHash
+          token
+          timestamp
+        }
+      }
+    `
+  }
+
+  _getDestinationTxGQL(transferId: string, recipient: string, isFromL1Source: boolean) {
+    return isFromL1Source
+      ? this.GQL_getDestinationTxHashFromL1Source(recipient)
+      : this.GQL_getDestinationTxHashFromL2Source(transferId)
+  }
+
+  _getHop(network: Network, signer = undefined) {
+    if (!network) return null
+    return new Hop(network === 'mainnet' ? 'mainnet' : 'kovan', signer)
+  }
+
+  _getAllTokens(hop:any) {
+    const bridge = hop.bridge('ETH')
+    const token = bridge.getCanonicalToken(hop.Chain.Ethereum)
+    return token.addresses
+  }
+
+  _getClient(network: Network, walletId: string, from: string, fromAccountId: string) {
+    return this.getClient(network, walletId, from, fromAccountId)
+  }
+
+  async _getSigner(network: Network, walletId: string, from: string, fromAccountId: string, provider: any) {
+    const client = this._getClient(network, walletId, from, fromAccountId)
+    const privKey = await client.wallet.exportPrivateKey()
+    return new Wallet(privKey, provider)
+  }
+
+  async _getBridgeWithSigner(hopAsset: any, hopChainFrom: any, network: Network, walletId: string, from: string, fromAccountId: string) {
+    const chainFrom = this.getChain(hopChainFrom.slug)
+    const client = this._getClient(network, walletId, from, fromAccountId)
+    const privKey = await client.wallet.exportPrivateKey()
+    const hop = this._getHop(network)
+    const signer = new Wallet(privKey, hop?.getChainProvider(chainFrom))
+    const bridge = hop?.connect(signer).bridge(hopAsset)
```

Comment:
All these question marks are making me worried. In this case, hop could be undefined and bridge could be undefined, but the code is triggered and will continue. Probably somewhere way before this code gets triggered, you can make some basic checks and exist early